invenio_rdm_records/github/release.py

```diff
@@ -0,0 +1,315 @@
+        # an "alternate" link.
+        if response.status_code == 300:
+            zipball_url = response.links.get("alternate", {}).get("url")
+            if zipball_url:
+                response = self.gh.api.session.head(zipball_url, allow_redirects=True)
+                # Another edge-case, is when the access token we have does not
+                # have the scopes/permissions to access public links. In that
+                # rare case we fallback to a non-authenticated request.
+                if response.status_code == 404:
+                    response = requests.head(zipball_url, allow_redirects=True)
+                    # If this response is successful we want to use the finally
+                    # resolved URL to fetch the ZIP from.
+                    if response.status_code == 200:
+                        zipball_url = response.url
+
+        assert (
+            response.status_code == 200
+        ), f"Could not retrieve archive from GitHub: {zipball_url}"
+
+    def load_citation_file(self):
+        """Returns the citation file data."""
+        citation_file_name = current_app.config.get("GITHUB_CITATION_FILE")
+        if not citation_file_name:
+            return {}
+
+        # Fetch the citation file and load it
+        content = self.retrieve_remote_file(citation_file_name)
+
+        data = (
+            yaml.safe_load(content.decoded.decode("utf-8"))
+            if content is not None
+            else None
+        )
+
+        return data
+
+    def resolve_record(self):
+        """Resolves an RDM record from a release."""
+        recid = self._retrieve_record_by_uuid(self.release_object.record_id)
+        return current_rdm_records_service.read(system_identity, recid.pid_value)
+
+    def _is_first_release(self):
+        """Checks whether the current release is the first release of the repository."""
+        return self.repository_object.releases.count() == 0
+
+    def _retrieve_recid_by_uuid(self, rec_uuid):
+        """Retrieves a persistent identifier given its objects uuid.
+
+        Helper function.
+        """
+        recid = PersistentIdentifier.get_by_object(
+            pid_type="recid",
+            object_uuid=rec_uuid,
+            object_type="rec",
+        )
+        return recid
+
+    def publish(self):
+        """Publish GitHub release as record.
+
+        Drafts and records are created using the current records service.
+        The following steps are run inside a single transaction:
+
+        - Create a draft.
+        - The draft's ownership is set to the user's id via its parent.
+        - Upload files to the draft.
+        - Publish the draft.
+
+        In case of failure, the transaction is rolled back and the release status set to 'FAILED'
+
+        :raises ex: any exception generated by the records service (e.g. invalid metadata)
+        """
+        try:
+            self.release_object.status = ReleaseStatus.PROCESSING
```

Comment:
Does this reflect on the DB without any session.add or commits?

Reply:
You are right, we have to commit after changing the status